flytekit/control_plane/tasks/task.py

```diff
@@ -0,0 +1,102 @@
+import uuid as _uuid
+
+from flytekit.common import interface as _interfaces
+from flytekit.common.core import identifier as _identifier
+from flytekit.common.exceptions import scopes as _exception_scopes
+from flytekit.common.mixins import hash as _hash_mixin
+from flytekit.configuration import internal as _internal_config
+from flytekit.engines.flyte import engine as _flyte_engine
+from flytekit.models import common as _common_model
+from flytekit.models import task as _task_model
+from flytekit.models.admin import common as _admin_common
+from flytekit.models.core import identifier as _identifier_model
+
+
+class FlyteTask(_hash_mixin.HashOnReferenceMixin, _task_model.TaskTemplate):
+    def __init__(self, type, metadata, interface, custom, container=None, task_type_version=0, config=None):
+        super(FlyteTask, self).__init__(
+            _identifier.Identifier(
```

Comment:
kill this, this should be a required argument to the constructor.  By the time someone is instantiating one of these objects, the ID should already exist and the user should know what it is.